feat(QML): add interface settings

acolombier opened the pull request:
Addresses #14539

JoergAtGithub commented:
This PR is huge and therefore difficult to review. Would it be possible to split in in one PR per tab?

I think the term "Interface" is too general, as a DJ software has so many other interfaces. I reccommend to keep the existing names without grouping, to make the transition for existing Mixxx users easier.

Does this introduce any new GUI strings to translate, or is ensured, that the existing translations are kept?

acolombier commented:
Yep, that's fair. 

I'm afraid the `Waveform` tab will still be pretty large though.

acolombier committed:
feat(QML): add interface settings

acolombier commented:
Rebased and fixed all conflict - this should be the next QML feature to look at. Note that the setting panel may currently look a little off, due to the missing changes around category loading (see other QML PR)

> Does this introduce any new GUI strings to translate, or is ensured, that the existing translations are kept?

Yes, it does! I have keep near 100% of all existing strings, taken directly from the `.ui` file. There may be one or two case where I had to merge or split a single string into multiple (or vice-versa) but this would be minor edge case.

Sorry, this has been sometime I worked on this, but I can confirm this was something I was being careful with when I done it.

acolombier commented:
I have slightly rework the section where the margin was the most problematic as well as the colour section which was too packed:

<img width="1238" height="387" alt="Screenshot From 2026-01-28 00-58-19" src="https://github.com/user-attachments/assets/e614f94e-25bf-4b42-b0cc-9c5721799cc5" />

Alignment and sizing is not perfect, but since I had to rework the layout in#15483, I'd like to keep change to a minimum to limit conflict and focus on this there or in a second pass. Are you happy with this approach @ronso0 ?

acolombier commented:
Aligned with @daschuer's comment [here](https://github.com/mixxxdj/mixxx/pull/13781#issuecomment-2476433284), it looks like the issue seems to come with clazy 1.12, but doesn't not impact the versions that comes in `ubuntu-24.04`, thus why I could not reproduce it (gcc-14 being installed in the Github action runner image).

Since 1.12 is not officially supported on Ubuntu, I have changed the customer PPA workaround to uninstalling gcc-14 instead in 7d141913912eb61b9904ff4a1c1591a31fce41a6
